File: src/lessons/recursion/sierpinski/SierpinskiEntity.scala
package lessons.recursion.sierpinski;

class ScalaSierpinskiEntity extends jlm.universe.turtles.Turtle {

	/* BEGIN TEMPLATE */
	def sierpinski(level:Int, length:Double) {
		/* BEGIN SOLUTION */
		if (level >= 0) {
			for (i <- 1 to 3) {
				forward(length / 2.);
				right(360. / 3.);
				sierpinski(level-1, length / 2.);
				left(360. / 3.);
				forward(length / 2.);
				right(360. / 3.);
			}
		}
		/* END SOLUTION */
	}
	/* END TEMPLATE */

	override def run() {
		sierpinski(getParam(0).asInstanceOf[Int], getParam(1).asInstanceOf[Double]);
	}

}

File: src/lessons/recursion/sierpinski/SierpinskiEntity.c
//RemoteTurtle

/* BEGIN TEMPLATE */
void sierpinski(int level, double length) {
	/* BEGIN SOLUTION */
	if (level >= 0) {
		int i;
		for (i = 0; i < 3; i++) {
			 sierpinski(level-1,length/2);
			 forward(length);
			 right(120);
		}
	}
	/* END SOLUTION */
}
/* END TEMPLATE */

void run() {
	sierpinski( getParamInt(0), getParamDouble(1));
}
